### ISSUE TITLE
Test workloads are no longer cleaned up after each test

### ISSUE THREAD

**Issue by kaovilai** [openshift/oadp-operator#1244]
https://github.com/openshift/oadp-operator/pull/1223

TLDR reportAfterEach which contained cleanup code is removed and not added back.

https://github.com/openshift/oadp-operator/pull/1223/files#diff-f7f90d1b403b9620176bb058724a225eacdbf0fb56df0c9f77f7cd4f3152f917L68

**Comment by kaovilai** [comment]
Seeing this moved into `func tearDownBackupAndRestore(`

**Comment by kaovilai** [comment]
```
	var _ = AfterEach(func(ctx SpecContext) {
		tearDownBackupAndRestore(lastBRCase, lastInstallTime, ctx.SpecReport())
	})
```
Being run here.

This should still not run on anyting that is skipped.

```
		if report.State == types.SpecStateSkipped || report.State == types.SpecStatePending {
			// do not run if the test is skipped
			return
		}
```
I'm not seeing SpecStateSkipped added

**Comment by mateusoliveira43** [comment]
I think this is not needed for AfterEach. Docs for reportAfterEach "ReportAfterEach nodes are run for each spec, even if the spec is skipped or pending."

**Comment by kaovilai** [comment]
not sure exactly what I spotted but it was along the lines of failures related to an existing workload deployment already exists blocking new creation in logs somewhere.

**Comment by kaovilai** [comment]
I will run e2e and check cleanup still works later.. and will close this if so

**Comment by mateusoliveira43** [comment]
example problem https://github.com/openshift/oadp-operator/pull/1239#issuecomment-1832048043

**Comment by kaovilai** [comment]
example 2 https://github.com/openshift/oadp-operator/pull/1218?notification_referrer_id=NT_kwDOAKtTeLM4MzMwNDc1NDc4OjExMjI4MDI0#issuecomment-1832515977

There are 9 results with `- 		"velero.io/backup-name":  string("`
which is part of the code that shows unexpected label diff with an existing workload that was not cleanup.

**Comment by kaovilai** [comment]
The following backup name label diff should not (ie. object in cluster should not already exists) occur on e2e-app creation if cleanup was done.
```
diff found for key: metadata
    map[string]any{
    	"creationTimestamp": string("2023-11-27T23:06:24Z"),
    	"generation":        int64(1),
    	"labels": map[string]any{
    		"e2e-app":                string("true"),
  - 		"velero.io/backup-name":  string("mongo-datamover-e2e-3ec695a1-8d79-11ee-8f5d-0a580a83a32f"),
  - 		"velero.io/restore-name": string("mongo-datamover-e2e-3ec695aa-8d79-11ee-8f5d-0a580a83a32f"),
```

**Comment by kaovilai** [comment]
Looks like we're not cleaning up SCC. This log kept popping up when installing apps
```
updating resource: security.openshift.io/v1, Kind=SecurityContextConstraints; name: mongo-persistent-scc
```

**Comment by sseago** [comment]
@kaovilai yeah, those are cluster-scoped, so namespace removal won't get rid of them.
